### PR TITLE
Draft Release Notes for Gradle 4.1

### PR DESCRIPTION
**Pull request by eriwen**
### Contributor Checklist
- [x] Animated GIF
- [ ] Gratuitous Emoji 🔥  ✨  🌈 
- [x] Initial review from @big-guy

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -1,8 +1,22 @@
-The Gradle team is pleased to announce the release of Gradle 4.1.
+The Gradle team is pleased to announce the first release candidate for Gradle 4.1.
 
-This release now supports running Gradle on the most recent JDK 9 release (b170+). It also optimizes startup speed, positively affecting the execution time of every build.
+First and foremost, we are excited to announce the new [Worker API](userguide/custom_tasks.html#worker_api) for Gradle plugins, facilitating **safe parallel execution**. We encourage plugin authors to [read the new user guide section](userguide/custom_tasks.html#worker_api) as we plan to leverage this API to enable parallel execution by default in a future release. The [Gradle plugin for Android 3.0 alpha](https://developer.android.com/studio/preview/features/new-android-plugin.html) has already adopted this API, so plugin developers will be in good company.
```
Comment (ghale) [inline subprojects/docs/src/docs/release/notes.md]: > as we plan to leverage this API to enable parallel execution by default in a future release

This reads wrong to me as it implies that parallel execution by default isn't already enabled.  In fact, one of the reasons to use this API is specifically to get default parallel execution.  What's not enabled right now is parallel execution of tasks that _do not_ use this API (although you can get parallelization of non-worker-api tasks from different projects using "--parallel").  Does that make sense?

Comment (ghale) [inline subprojects/docs/src/docs/release/notes.md]: I would reword this to something like "We encourage plugin authors to [read the new user guide section](userguide/custom_tasks.html#worker_api) and implement tasks using this API as this enables safe parallel execution of tasks."

**Comment by jjohannes** [comment]
Great preamble @eriwen! I Like it!

Minor remark about the notes: I would consider removing the **Promoted features** section as there are none.

Or, if we always keep that in for consistency, change 

> The following are the features that have been promoted in this Gradle release.

to

> There are no promoted features in this release.

(We should probably also add this sentence as an option to the template. This section has been empty in the past as well.)

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -1,8 +1,22 @@
-The Gradle team is pleased to announce the release of Gradle 4.1.
+The Gradle team is pleased to announce the first release candidate for Gradle 4.1.
```
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md]: I think we always used the removed sentence. The reason is that we will then no need to change the notes between RCs. Has there been some discussion to change this to always change the release notes between RCs and the final release?

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -166,24 +175,23 @@ The `--tests` filters are now always applied on top of the filtering defined in
...
- - [Bo Zhang](https://github.com/blindpirate) - Contributions to consistent --tests option handling (#2172)
- - [Marcin Zajączkowski](https://github.com/szpak) - Add `@since` tag to `Project.findProperty()` (#2403)
- - [Seth Jackson](https://github.com/sethjackson) - Fix the default daemon JVM args on Java 8 (#2310)
- - [Ismael Juma](https://github.com/ijuma) - Update default Zinc compiler version to 0.3.15 with preliminary Java 9 support (#2420)
- - [Krzysztof Ropiak](https://github.com/krro) - Sets annotation processor classpath in Scala compilation task (#2281)
- - [Dave Brewster](https://github.com/dbrewster) - Add support for caching Scala compilation (#1958)
- - [Steve Dougherty](https://github.com/Thynix) - Prevent spurious gradle-wrapper.jar changes (#2183)
- - [Yannick Welsch](https://github.com/ywelsch) - Use GNU-style release flag for Java 9 compiler (#2474)
- - [Mike Kobit](https://github.com/mkobit) - Add Worker API classes to JavaDocs (#2372)
+ - [Dave Brewster](https://github.com/dbrewster) - Add support for caching Scala compilation (gradle/gradle#1958)
```
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md]: Should we sort the list of contributors in any way? I remember sorting them alphabetically for some earlier release. Now the order just seems to be random.

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: Reverted, but I'd like to keep this RC language for GitHub because the releases are separate.

File: subprojects/docs/src/docs/release/notes.md (hunk 3 above)
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: I sorted best I could in terms of impact. More impactful PRs at the top.

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: Reworded. Please take a look.

**Comment by eriwen** [comment]
@jjohannes Feedback addressed. We cannot remove that section as it'll fail the build.

File: subprojects/docs/src/docs/release/notes.md (hunk 3 above)
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md]: I think it would be more transparent and easy to do if we would just sort alphabetical. Your call.
